api/src/main/java/org/ehcache/spi/serialization/Serializer.java

```diff
@@ -26,7 +26,20 @@
 
 /**
  * Interface defining the contract used to transform types in a serial form.
- * Implementations must be thread-safe and must define a constructor accepting a single ClassLoader parameter.
+ * 
+ * When used within the DefaultSerializationProvider which is the default implementation of {@link SerializationProvider} 
+ * service provided by ehcache, there are a few additional requirements:
+ * The implementations must be thread-safe and must define either or both of the two constructors:
+ * <ul>
+ *   <li>
+ *     Accepting a single ClassLoader parameter: This constructor will be used to initialize the serializer for
+ *     transient caches
+ *   </li>
+ *   <li>
+ *     Accepting a ClassLoader parameter and a FileBasedPersistenceContext: This constructor will be used to initialize
+ *     the serializer for persistent caches 
```

Comment:
I am assuming that the previous comment is applicable to to the `FileBasedPersistenceContext ` reference as well